ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.ts

```diff
@@ -1,88 +1,203 @@
 import {
     Component,
+    ElementRef,
     EventEmitter,
     Input,
     OnDestroy,
-    Output
+    OnInit,
+    Output,
+    ViewChild
 } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
 
 import { EMPTY, Subject } from 'rxjs';
-import { catchError, takeUntil } from 'rxjs/operators';
+import { catchError, finalize, takeUntil } from 'rxjs/operators';
 
-import { FileUploadBody, FileUploadResponse } from '../../../../models/fileUpload';
+import { UploadFile } from '../../../../models/uploadFile';
 import { ActivityFileQuestionBlock } from '../../../../models/activity/activityFileQuestionBlock';
 import { FileUploadService } from '../../../../services/fileUpload.service';
 import { LoggingService } from '../../../../services/logging.service';
+import { ActivityFileAnswerDto } from '../../../../models/activity/activityFileAnswerDto';
+import { FileUploadResponse } from '../../../../models/fileUploadResponse';
+import { ModalDialogService } from '../../../../services/modal-dialog.service';
+import { ConfirmDialogComponent } from '../../../confirmDialog/confirmDialog.component';
+import { ActivityFileValidationRule } from '../../../../services/activity/validators/activityFileValidationRule';
+import { FileAnswerMapperService } from '../../../../services/fileAnswerMapper.service';
 
 @Component({
     selector: 'ddp-activity-file-answer',
     templateUrl: 'activityFileAnswer.component.html',
     styleUrls: ['activityFileAnswer.component.scss']
 })
-export class ActivityFileAnswer implements OnDestroy {
+export class ActivityFileAnswer implements OnInit, OnDestroy {
     @Input() block: ActivityFileQuestionBlock;
     @Input() readonly: boolean;
     @Input() studyGuid: string;
     @Input() activityGuid: string;
     @Output() valueChanged: EventEmitter<string | null> = new EventEmitter();
-
-    file: File;
-    fileUploadData: FileUploadResponse;
-    isFileSelected: boolean;
+    @Output() componentBusy = new EventEmitter<boolean>();
+    @ViewChild('uploaded', {read: ElementRef}) private uploadedFileRef: ElementRef;
+    @ViewChild('undoUploadBtn', {read: ElementRef}) private undoUploadButtonRef: ElementRef;
+    fileToUpload: UploadFile | null;
+    uploadedFile: ActivityFileAnswerDto | null;
+    errorMessage: string;
+    isLoading: boolean;
     private ngUnsubscribe = new Subject();
+    private readonly panelClass = 'file-upload-confirm-dialog';
 
     constructor(private fileUploadService: FileUploadService,
-                private logger: LoggingService) {
+                private logger: LoggingService,
+                private dialog: MatDialog,
+                private modalDialogService: ModalDialogService) {
+    }
+
+    ngOnInit(): void {
+        this.initUploadedFile();
     }
 
     onFilesSelected(files: File[]): void {
```

Comment:
I am getting a type warning in template. Change below gets rid of it.
```suggestion
    onFilesSelected(files: File[] | FileList): void {
```

Reply:
Fixed at f929b71f78f178edac82e717e16d6cc391afd774